### ISSUE TITLE
[$1000] DEV - Console errors when opening a chat

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#24417]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Run the web version of the app i.e. `npm run web`.
2. Open Chrome developer tools before the app fully loads and switch to the console tab.
## Expected Result: 
There are no console errors.
## Actual Result: 
There are a couple of errors with the topmost one stating, "Warning: Failed prop type: Invalid prop `personalDetails.accountID` of type of `number` supplied to `WithCurrentUserPersonalDetails(Component)`, expected `object`.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 
**Reproducible in staging?:** dev 
**Reproducible in production?:** dev
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![expensify-dev-console-errors](https://github.com/Expensify/App/assets/43996225/af37d7a4-529e-4770-bf44-92065959c177)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Victor-Nyagudi 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1691050656767699

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ef7804b7f81bc7fe</li>
        <li>Upwork Job ID: 1690070386539089920</li>
        <li>Last Price Increase: 2023-08-11</li>
    </ul>
</details>

**Comment by jfquevedol2198** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
DEV - Console errors when opening a chat

### What is the root cause of that problem?
`accountID` of `personalDetailsPropType` is defined as `number`, but `string` type is passed.

### What changes do you think we should make in order to solve the problem?
https://github.com/Expensify/App/blob/c08d5887b18914d95e64e925bfed290fecfffc94/src/pages/personalDetailsPropType.js#L20

Update `accountID` type like this:
```javascript
accountID: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by lschurr** [comment]
@sobitneupane could you review and confirm this is reproducible?

**Comment by Talha345** [comment]
There are quite a few errors that appear on different actions and mostly are type errors which will be mitigated once the TS migration is done so I wonder if we really need a separate task for this?

**Comment by situchan** [comment]
Dupe of https://github.com/Expensify/App/issues/23645

**Comment by lschurr** [comment]
Closing since this is a dupe and the other bug was reported earlier.
